src/game_inventory.cpp

```diff
@@ -1383,14 +1383,17 @@ void game_menus::inv::insert_items( avatar &you, item_location &holster )
         item &it = *holstered_item.first;
         bool success = false;
         if( !it.count_by_charges() ) {
-            if( all_pockets_rigid || holster.parents_can_contain_recursive( &it ) ) {
+            if( ( all_pockets_rigid || holster.parents_can_contain_recursive( &it ) ) &&
+                holster.get_item()->best_pocket( it, holster ).second != nullptr ) {
```

Comment:
I'd put the check first in that statement, because checking parents is unnecessary when the holster can't contain the item anyway.

Reply:
thank you that makes sense